diff --git a/CMakeLists.txt b/CMakeLists.txt
index a50e673d..dd08e3cc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -49,7 +49,7 @@ if (CMP_HOST_WINDOWS)
     add_compile_options(/W4 /wd4201)
 #    add_compile_options(/W4 /WX /wd4201)
 elseif(CMP_HOST_LINUX)
-    add_compile_options(-Wall)
+#    add_compile_options(-Wall)
 #    add_compile_options(-Wall -Werror)
 endif()
 
diff --git a/cmp_core/CMakeLists.txt b/cmp_core/CMakeLists.txt
index 47873770..41805e40 100644
--- a/cmp_core/CMakeLists.txt
+++ b/cmp_core/CMakeLists.txt
@@ -63,45 +63,3 @@ if (UNIX)
 endif()
 
 set_target_properties(CMP_Core PROPERTIES FOLDER ${PROJECT_FOLDER_SDK_LIBS})
-
-# Core SIMD options
-
-# SSE
-add_library(CMP_Core_SSE STATIC)
-target_sources(CMP_Core_SSE PRIVATE source/core_simd_sse.cpp)
-target_include_directories(CMP_Core_SSE PRIVATE source shaders)
-
-if (UNIX)
-    target_compile_options(CMP_Core_SSE PRIVATE -march=nehalem)
-endif()
-
-set_target_properties(CMP_Core_SSE PROPERTIES FOLDER ${PROJECT_FOLDER_SDK_LIBS})
-
-# AVX
-add_library(CMP_Core_AVX STATIC)
-target_sources(CMP_Core_AVX PRIVATE source/core_simd_avx.cpp)
-target_include_directories(CMP_Core_AVX PRIVATE source shaders)
-
-if (WIN32)
-    target_compile_options(CMP_Core_AVX PRIVATE /arch:AVX2)
-else()
-    target_compile_options(CMP_Core_AVX PRIVATE -march=haswell)
-endif()
-
-set_target_properties(CMP_Core_AVX PROPERTIES FOLDER ${PROJECT_FOLDER_SDK_LIBS})
-
-# AVX-512
-add_library(CMP_Core_AVX512 STATIC)
-target_sources(CMP_Core_AVX512 PRIVATE source/core_simd_avx512.cpp)
-target_include_directories(CMP_Core_AVX512 PRIVATE source shaders)
-
-if (WIN32)
-    target_compile_options(CMP_Core_AVX512 PRIVATE /arch:AVX-512)
-else()
-    target_compile_options(CMP_Core_AVX512 PRIVATE -march=knl)
-endif()
-
-set_target_properties(CMP_Core_AVX512 PROPERTIES FOLDER ${PROJECT_FOLDER_SDK_LIBS})
-
-# Link SIMD libraries to CMP_Core
-target_link_libraries(CMP_Core PRIVATE CMP_Core_SSE CMP_Core_AVX CMP_Core_AVX512)
\ No newline at end of file
diff --git a/cmp_core/shaders/bc1_cmp.h b/cmp_core/shaders/bc1_cmp.h
index c97315dd..2b3dacd7 100644
--- a/cmp_core/shaders/bc1_cmp.h
+++ b/cmp_core/shaders/bc1_cmp.h
@@ -109,22 +109,8 @@ CMP_STATIC bool bc1ToggleSIMD(CGU_INT newExtension)
         useSSE42  = newExtension == EXTENSION_SSE42;
     }
 
-    if (useAVX512 && IsAvailableAVX512(extensions))
-    {
-        cpu_bc1ComputeBestEndpoints = avx512_bc1ComputeBestEndpoints;
-    }
-    else if (useAVX2 && IsAvailableAVX2(extensions))
-    {
-        cpu_bc1ComputeBestEndpoints = avx_bc1ComputeBestEndpoints;
-    }
-    else if (useSSE42 && IsAvailableSSE4(extensions))
-    {
-        cpu_bc1ComputeBestEndpoints = sse_bc1ComputeBestEndpoints;
-    }
-    else
-    {
-        cpu_bc1ComputeBestEndpoints = _cpu_bc1ComputeBestEndpoints;
-    }
+
+    cpu_bc1ComputeBestEndpoints = _cpu_bc1ComputeBestEndpoints;
 
     g_bc1FunctionPointersSet = true;
 